frontend/ui/uimanager.lua

```diff
@@ -47,8 +48,42 @@ function UIManager:init()
         -- resume.
         self:_initAutoSuspend()
         self.event_handlers["Suspend"] = function(input_event)
-            self:_stopAutoSuspend()
-            Device:onPowerEvent(input_event)
+            if not self._power_off_triggered then
```

Comment:
If `_power_off_triggered` is only introduced to prevent race from suspend and poweroff, we can avoid it by just nuking `self.event_handlers["Suspend"]` in `ok_callback`.